### PR TITLE
CEDS-1906 Extract origination country to new page

### PR DESCRIPTION
**Pull request by PatrykRudnicki**

### CODE REVIEW

File: app/controllers/declaration/OriginationCountryController.scala [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+    DestinationCountries
+      .form(OriginationCountryPage)
+      .bindFromRequest()
+      .fold(
+        formWithErrors => Future.successful(BadRequest(originationCountryPage(mode, formWithErrors))),
+        validCountry => {
+          val updatedCache =
+            request.cacheModel.locations.destinationCountries.getOrElse(DestinationCountries.empty()).copy(countryOfDispatch = validCountry)
+
+          updateExportsDeclarationSyncDirect(model => model.copy(locations = model.locations.copy(destinationCountries = Some(updatedCache)))).map {
```
Comment (bambuchaAdm) [inline app/controllers/declaration/OriginationCountryController.scala]: please move copy as method

File: app/views/declaration/destinationCountries/origination_country.scala.html [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+
+    @components.heading(
+        title = messages("declaration.originationCountry.question"),
+        sectionHeader = messages("declaration.originationCountry.heading")
+    )
+
+    @helper.form(routes.OriginationCountryController.submit(), 'autoComplete -> "off") {
+        @helper.CSRF.formField
+
+        <div id="origination-country-field">
```
Comment (bambuchaAdm) [inline app/views/declaration/destinationCountries/origination_country.scala.html]: why this id is on div not on `field_autocomplete`

File: test/views/declaration/destinationCountries/OriginationCountryViewSpec.scala [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+import unit.tools.Stubs
+import views.declaration.spec.UnitViewSpec
+import views.html.declaration.destinationCountries.origination_country
+
+class OriginationCountryViewSpec extends UnitViewSpec with Stubs {
+
+  val form: Form[String] = DestinationCountries.form(OriginationCountryPage)
+
+  val originationCountryPage = new origination_country(mainTemplate)
+  val view = originationCountryPage(Mode.Normal, form)
```
Comment (bambuchaAdm) [inline test/views/declaration/destinationCountries/OriginationCountryViewSpec.scala]: Please test this for each declaration type

File: app/controllers/declaration/OriginationCountryController.scala (hunk 1 above)
Comment (PatrykRudnicki) [reply app/controllers/declaration/OriginationCountryController.scala]: Changed

File: app/views/declaration/destinationCountries/origination_country.scala.html (hunk 2 above)
Comment (PatrykRudnicki) [reply app/views/declaration/destinationCountries/origination_country.scala.html]: It's the same like other usage of autocomplete input. I don't want to do it with this PR but I'll keep in mind to move id to autocomplete

File: test/views/declaration/destinationCountries/OriginationCountryViewSpec.scala (hunk 3 above)
Comment (PatrykRudnicki) [reply test/views/declaration/destinationCountries/OriginationCountryViewSpec.scala]: This view doesn't change and it's not related with any specific declaration type. I'll change behaviour of all of this, to align with specific journeys, this is just starting point for everything. It'll be added in the future

**Pull request merged**